cooking-app/src/components/Content/ListRecipes/ListRecipes.js

```diff
@@ -0,0 +1,138 @@
+import { onSnapshot } from "firebase/firestore";
+import { useEffect, useRef, useState } from "react";
+import { recipesCollection } from "../../../api/firebaseIndex";
+import { IndividualRecipe } from "./IndividualRecipe";
+import { getDataFromSnapshot } from "../../../utils/GetDataFromSnapshot";
+import { PageTitle } from "../../styles/Global.styled";
+import styled from "styled-components";
+
+export const ListRecipes = () => {
+  const [datafromFirebase, setdatafromFirebase] = useState([]);
+  const [search, setSearch] = useState("");
+  const [salt, setSalt] = useState(false);
+  const [sweet, setSweet] = useState(false);
+  const [breakfast, setBreakfast] = useState(false);
+  const [dinner, setDinner] = useState(false);
+  const [dessert, setDessert] = useState(false);
+  const [lunch, setLunch] = useState(false);
+
+  const textInput = useRef();
+  const saltInput = useRef();
+  const sweetInput = useRef();
+  const breakfastInput = useRef();
+  const dinnerInput = useRef();
+  const dessertInput = useRef();
+  const lunchInput = useRef();
+
+  useEffect(() => {
+    onSnapshot(recipesCollection, (snapshot) => {
+      setdatafromFirebase(getDataFromSnapshot(snapshot));
+    });
+  }, []);
+
+  const listofRecipe2 = datafromFirebase
+    .filter((item) => {
+      if (salt == true) {
+        return item.categories.includes("Salt");
+      } else if (sweet === true) {
+        return item.categories.includes("Sweet");
+      } else if (breakfast === true) {
+        return item.categories.includes("Breakfast");
+      } else if (dinner === true) {
+        return item.categories.includes("Dinner");
+      } else if (lunch === true) {
+        return item.categories.includes("Lunch");
+      } else if (dessert === true) {
+        return item.categories.includes("Dessert");
+      } else if (search.toLowerCase() === "") {
+        return item;
+      } else return item.title.toLowerCase().includes(search);
+    })
+    .map((singleRecipe) => {
+      return <IndividualRecipe singleRecipe={singleRecipe} />;
+    });
```

Comment:
Pomyślcie jak napisać to lepiej, bez tylu powtórzeń? Jak chcecie możemy zastanowić się razem.